### ISSUE TITLE
hunyuanvideo/videoalign reward curve

### ISSUE THREAD

**Issue by MuteApo** [XueZeyue/DanceGRPO#30]
hi zeyue，我这边在尝试复现hunyuanvideo在videoalign上的曲线，但是32卡上的结果和readme中贴出的相差很大，reward几乎不升，长期来看甚至在下降，也未见反弹的迹象：

<img width="418" height="301" alt="Image" src="https://github.com/user-attachments/assets/c35eedb8-7f6a-4010-ab57-c2f3688195c9" />

而readme中32卡是稳定上升的，即使16卡训练，虽然早期也会下降，但能在70iter左右观察到强劲反弹：

<img width="421" height="639" alt="Image" src="https://github.com/user-attachments/assets/f139e5b0-5170-4597-91f7-8aee58366302" />

我目前的setting尽可能参照了各位大佬在issue里的讨论，prompt=32，rollout=12，bestofn=8，sampling_steps=16，timestep_fraction=0.6，mq_coef=0.0，想知道我的setting是否已经跟readme中展示的曲线对齐：

```shell
torchrun \
    --nnodes=$HOST_NUM \
    --node_rank=$INDEX \
    --nproc_per_node=$HOST_GPU_NUM \
    --master_addr=$CHIEF_IP \
    --master_port=29500 \
    fastvideo/train_grpo_hunyuan.py \
    --seed 42 \
    --model_type "hunyuan_hf" \
    --pretrained_model_name_or_path data/HunyuanVideo \
    --vae_model_path data/HunyuanVideo \
    --cache_dir data/.cache \
    --data_json_path data/rl_embeddings/videos2caption.json \
    --validation_prompt_dir data/Mochi-Black-Myth/validation \
    --gradient_checkpointing \
    --train_batch_size 1 \
    --sp_size 1 \
    --train_sp_batch_size 1 \
    --dataloader_num_workers 4 \
    --gradient_accumulation_steps 4 \
    --max_train_steps 202 \
    --learning_rate 1e-5 \
    --mixed_precision bf16 \
    --checkpointing_steps 10 \
    --validation_steps 100000000 \
    --validation_sampling_steps 8 \
    --checkpoints_total_limit 3 \
    --allow_tf32 \
    --ema_start_step 0 \
    --cfg 0.0 \
    --ema_decay 0.999 \
    --log_validation \
    --output_dir checkpoints/gen12_best8_frac0.6 \
    --h 480 \
    --w 480 \
    --t 53 \
    --sampling_steps 16 \
    --eta 0.25 \
    --lr_warmup_steps 0 \
    --fps 8 \
    --sampler_seed 1237 \
    --max_grad_norm 1.0 \
    --weight_decay 0.0001 \
    --num_generations 12 \
    --shift 5 \
    --use_group \
    --use_videoalign \
    --timestep_fraction 0.6 \
    --use_same_noise \
    --bestofn 8 \
    --vq_coef 1.0 \
    --mq_coef 0.0 \
```

代码方面目前没做任何修改，如果有其他可能的原因，也请zeyue帮忙check一下，感谢！

**Comment by XueZeyue** [comment]
@MuteApo hi，问题解决了吗

**Comment by k15201363625** [comment]
感谢作者的开源，以及issue积极回答各种问题！
同样的问题想请教，但使用了8张卡，并增大了batch size以保证总的batchsize不太小（4*2*8），未改变参数设置（num_generation=24 bestofn=8）。
在 原始数据集，1000 moviegen数据集，with/without lora（rank 256 alpha 128）共4个版本，等尝试上目前100+steps均未看到vq reward 上升，请问还有什么参数调整的经验吗😭
比如：
总batchsize更大会更稳定吗
lora会导致reward更容易下降吗
hwt重要吗（之前reward tuning wan发现resolution设置为256会炸 需要原始分辨率）
kl divergence加上是不是会稳定点
grpo training cfg 设置6018.0 会有关系吗
use_same_noise 打开或关闭哪种会更好

或者有其他training不动的经验，感谢分享！ 


(下面的reward 未经scale)
<img width="1089" height="569" alt="Image" src="https://github.com/user-attachments/assets/58c2bfbb-436b-4675-a159-c1e3de8e811c" />
